subsys/logging/log_output.c

```diff
@@ -162,6 +162,10 @@ static void buffer_write(log_output_func_t outf, uint8_t *buf, size_t len,
 
 void log_output_flush(const struct log_output *output)
 {
+	if (!output->control_block->offset) {
+		return;
+	}
+
 	buffer_write(output->func, output->buf,
```

Comment:
what about changing `buffer_write()` function.
From
```
do {
   ...
} while (len != 0);
```
To
```
while(len != 0) {
   ...
}
```

Reply:
Good point. Get patch updated.